installers/Config.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE RecordWildCards   #-}
+module Config
+  ( generateConfig
+  , Request(..)
+  , OS(..), Cluster(..), Config(..)
+  ) where
+
+import qualified Control.Exception
+
+import qualified Data.Char
+import qualified Data.ByteString
+import           Data.Semigroup
+import           Data.Text.Lazy
+import qualified Data.Yaml
+
+import qualified Dhall
+import qualified Dhall.Import
+import qualified Dhall.JSON
+import qualified Dhall.Parser
+import qualified Dhall.TypeCheck
+
+import qualified GHC.IO.Encoding
+
+import qualified System.IO
+import qualified System.Exit
+
+import           Text.Trifecta.Delta (Delta(..))
+
+import Prelude
+
+data OS
+  = Linux
+  | Macos64
+  | Win64
+  deriving (Eq, Show)
+
+data Cluster
+  = Mainnet
+  | Staging
+  deriving (Eq, Read, Show)
+
+data Config
+  = Launcher
+  | Topology
+  deriving (Eq, Show)
+
+data Request
+  = Request
+  { rOS      :: OS
+  , rCluster :: Cluster
+  , rConfig  :: Config
+  } deriving (Eq, Show)
+
+lshow :: Show a => a -> Text
+lshow = Data.Text.Lazy.pack . fmap Data.Char.toLower . show
+
+dhallTopExpr :: Text -> Config -> OS -> Cluster -> Text
+dhallTopExpr path Launcher os cluster = path <> "/launcher.dhall ( "<>path<>"/" <> lshow cluster <> ".dhall "<>path<>"/" <> lshow os <> ".dhall ) "<>path<>"/" <> lshow os <> ".dhall"
+dhallTopExpr path Topology os cluster = path <> "/topology.dhall ( "<>path<>"/" <> lshow cluster <> ".dhall "<>path<>"/" <> lshow os <> ".dhall )"
+
+generateConfig :: Request -> FilePath -> FilePath -> IO ()
+generateConfig Request{..} configRoot outFile = handle $ do
+  GHC.IO.Encoding.setLocaleEncoding GHC.IO.Encoding.utf8
+
+  Dhall.detailed $ do
+    let inText = dhallTopExpr (Data.Text.Lazy.pack configRoot) rConfig rOS rCluster
+    print inText
+    expr <- case Dhall.Parser.exprFromText (Directed "(stdin)" 0 0 0 0) inText of
+              Left  err  -> Control.Exception.throwIO err
+              Right expr -> return expr
+
+    expr' <- Dhall.Import.load expr
+    case Dhall.TypeCheck.typeOf expr' of
```

Comment:
Pretty sure this is not needed? `dhallToJSON` will definitely type check

Reply:
@domenkozar, it doesn't: https://github.com/dhall-lang/dhall-json/blob/master/src/Dhall/JSON.hs#L159

Reply:
`Dhall.Import.load` does call typeOf.

Reply:
@domenkozar, are you referring to the import-from-URL edge case in  https://github.com/dhall-lang/dhall-haskell/blob/master/src/Dhall/Import.hs#L652 ?

Reply:
Ah I see - you copied most of this code from https://github.com/dhall-lang/dhall-json/blob/master/dhall-to-json/Main.hs#L35-L74 - maybe we should submit a PR to dhall-to-json to be able to use it as a haskell library?

Reply:
@domenkozar, sure thing!

Reply:
https://github.com/dhall-lang/dhall-json/pull/20 is WIP, pending feedback from Gabriel.

Reply:
..merged!